tutorials/abap-environment-create-rap-business-events/abap-environment-create-rap-business-events.md

```diff
@@ -0,0 +1,543 @@
+ - You need a user with access to maintain communication arrangement.
+
+## Details
+### You will learn
+  - How to create RAP Events Business
+  - How to set up a channel to connect to SAP Event Mesh
+
+>Always replace `####` with your initials or group number.
+
+---
+
+[ACCORDION-BEGIN [Step 1: ](Overview)]
+
+The ABAP RESTful Application Programming Model (RAP) now supports the native consumption and exposure of business events. For exposure, an event can be defined and raised in a RAP business object or in the behavior extension and then published via Event Bindings.
+
+[DONE]
+[ACCORDION-END]
+
+[ACCORDION-BEGIN [Step 2: ](Create database table)]
+
+To produce and raise an event you need first to define your RAP Business Object which produce the event. For this, you will create a simple online shop application. The event will be sent whenever a new order is created.
+
+  1. Open ADT and open your SAP BTP ABAP environment system.
+
+  2. If you do not have an **ABAP Package** create a new one. Please be sure if your package name is with **Z** like
+
+    - Name: `ZEVENT_EXPOSURE_####`
+    - Description: `define a RAP event`
+
+    ![new](2-1.png)
+
+  3. Right-click on your package and create a new database table
+
+    - Name: `ZONLINESHOP_####`
+    - Description: `DB for onlineshop`
+
+    ![database](2-2.png)
+
+    ![database](2-3.png)
+
+  4. Copy the code below to the Database and replace `####` with your number
+
+
+    ```ZONLINESHOP_####
+      define table zonlineshop_#### {
+      key client     : abap.clnt not null;
+      key order_uuid : sysuuid_x16 not null;
+      order_id       : abap.char(10) not null;
+      ordereditem    : abap.char(10) not null;
+      deliverydate   : abap.dats;
+      creationdate   : abap.dats;
+      }
+
+    ```
+
+  5. Save and activate your table.
+
+    ![database](2-4.png)
+
+[DONE]
+[ACCORDION-END]
+
+[ACCORDION-BEGIN [Step 3: ](Create CDS data model and projection view)]
+
+
+  1. Right-click on your package and create a data definition
+
+    - Name: `ZI_ONLINE_SHOP_####`
+    - Description: `data model for online shop`
+
+    ![cds](2-5.png)
+
+    ![cds](2-6.png)
+
+  2. Copy the code below to your CDS data model and replace `####` with your number
+
+
+    ```ZI_ONLINE_SHOP_####
+       define root view entity zi_online_shop_#### as select from zonlineshop_#### {
+       key order_uuid as Order_Uuid,
+       order_id as Order_Id,
+       ordereditem as Ordereditem,
+       deliverydate as Deliverydate,
+       creationdate as Creationdate  
+      }
+    ```
+
+  3. Save and activate your data model.
+
+    ![cds](2-7.png)
+
+  4. Right-click on your package and create a new data definition
+
+    - Name: `ZC_ONLINE_SHOP_####`
+    - Description: `projection view for online shop`
+
+    ![def](2-8.png)
+
+    after choosing a transport request click **Next** and choose **Define Projection View**
```

Comment:
Shouldn't this instruction have a number preceding it?  It is very clearly a task in the step.